### PR TITLE
New data model + APIs

### PR DESCRIPTION
**Pull request by brandonros**
Part of https://github.com/Banno/team-nodejs/issues/21

### CODE REVIEW

File: bin/mock-vault-data.sh [hunk 1]
```diff
@@ -5,6 +5,12 @@ API_GATEWAY_PUBLIC_KEY=$(cat mocks/api-gateway-public-key.pem)
 API_GATEWAY_PRIVATE_KEY=$(cat mocks/api-gateway-private-key.pem)
 OPENID_KEYS=$(cat mocks/openid-keys.json)
 
+# Enable transit engine
+vault secrets enable transit
+
+# Create transit encryption/decryption keys
+vault write -f transit/keys/node-sso-formatter-client_secret
```
Comment (brandonros) [inline bin/mock-vault-data.sh]: - [ ] Need an `environments` PR for this

File: src/controllers/bdi.ts [hunk 2]
```diff
@@ -5,15 +5,15 @@ import { buildForm } from '../lib/form'
...
+  const { decodedIdToken, client, configurationValues } = await buildSsoContext(req, 'BDI')
   const requiredClaims = [
     'cif',
     'email',
     'name',
     'address'
   ]
   validateClaims(requiredClaims, decodedIdToken)
-  const ssoUrl = client.redirectUris[0]
+  const ssoUrl = configurationValues.url
```
Comment (brandonros) [inline src/controllers/bdi.ts]: @ChadKillingsworth Cramming the SSO redirect URL into the OpenID client `redirectUris` model gave me issues.

You gave me conflicting requirements.

You told me "put the SSO URL for the vendor as redirectUris[0]`

But, `redirectUris[0]` needs to be `https://${enviornment}.banno.com/a/consumer/api/sso/institutions/${req.body.institutionId}/${req.body.name}` in my eyes.

Can you help clarify?

File: src/controllers/clients.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import * as express from 'express'
```
Comment (brandonros) [inline src/controllers/clients.ts]: ChadK does *not* want `node-sso-formatter` calling `node-openid-provider` internal APIs. Instead, he wants to duplicate the data and have Banno People POST to 2 places. I'm not a huge fan of this, but I also don't know his reasoning behind it.

File: src/lib/mocks.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
+/* istanbul ignore file */
```
Comment (brandonros) [inline src/lib/mocks.ts]: @ChadKillingsworth @eheikes How close does this get us to https://github.com/Banno/node-sso-formatter/issues/25 ? In my eyes, we can train Robin's team to mock all of their data.

I am able to test MFN SSO locally given this flow and would be more than willing to help any developer set up mock data for any other SSO. It's easier than juggling expiring JWTs + codes in my eyes...

Comment (eheikes) [reply src/lib/mocks.ts]: Good point. I haven't work with node-sso-formatter enough to answer you now, but it's certainly something to consider.

**Pull request merged**